pkg/auth/asynccache.go

```diff
@@ -27,13 +31,21 @@ func NewAsyncCache[T any](ctx context.Context, reloadPeriod time.Duration, cachi
 		cachingFunc:  cachingFunc,
 	}
 
-	item, err := cachingFunc(ctx)
+	var err error
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Minute)
```

Comment:
I would expect something in terms of seconds, not minutes :eyes: 
Remember that this blocks proper startup.